### ISSUE TITLE
Questions about batchnorm 

### ISSUE THREAD

**Issue by wlaud1001** [Han-Jia/UNICORN-MAML#3]
Hi, thank you for your great work!

I already saw the [past issue](https://github.com/Han-Jia/UNICORN-MAML/issues/2#issue-1231943195) about batchnorm, but i'm still little confused.

First of all, before you evaluate, [you set the encoder attribute 'is_training' to True](https://github.com/Han-Jia/UNICORN-MAML/blob/1681e4a72e7b240bb48b0dd66c993d3874919b9c/model/trainer/fsl_trainer.py#L177) and i don't know why you set that attribute to True and how it affects. 
As i know, there is no attribute named 'is_training' in PyTorch Module (and also in your encoder) and setting 'is_training' to True does not change the mode of Module. 
So if your purpose was to change the mode of encoder to 'train', i think using self.model.encoder.train() is the right way.

Secondly, if your purpose was to set the mode of encoder to 'train' and it works alright, then encoder will never use it's running mean and variance both training and evaluation phase. And then, there is no reason to [store and reset those values.](https://github.com/Han-Jia/UNICORN-MAML/blob/1681e4a72e7b240bb48b0dd66c993d3874919b9c/model/trainer/fsl_trainer.py#L236)

**Comment by wlaud1001** [comment]
And i also checked your earlier results, [AVITOR](https://github.com/Han-Jia/AVIATOR), and there exists 'is_training' in [resnet_maml](https://github.com/Han-Jia/AVIATOR/blob/58cdac7e32f25fed368b4f20086e61876ca12ac8/model/networks/resnet_maml.py#L74). 
Is there any reason why you removed 'is_training' in resnet?

**Comment by Han-Jia** [comment]
Hello,

In this code, the training state of the resnet is controlled by 'self.training'. So 'model.train()' and 'model.eval' can switch between two states. [Setting 'is_training' to True](https://github.com/Han-Jia/UNICORN-MAML/blob/1681e4a72e7b240bb48b0dd66c993d3874919b9c/model/trainer/fsl_trainer.py#L177) should be deleted and it will not influence the results.

In each task during meta-test, we use the 'forward_eval' function and [set the training state]([is_training' to True](https://github.com/Han-Jia/UNICORN-MAML/blob/1681e4a72e7b240bb48b0dd66c993d3874919b9c/model/models/maml.py#L56)) when the model is fine-tuned on the support set, and then [set the evaluation state](https://github.com/Han-Jia/UNICORN-MAML/blob/1681e4a72e7b240bb48b0dd66c993d3874919b9c/model/models/maml.py#L60) when we use the model to predict all instances on the query set. The BN is updated on the support set and we use the updated running statistics for query set evaluation. Since the running statistics of BN are reset in each task, the statistics of the previous task will not be shared.

**Comment by wlaud1001** [comment]
Oh, thank you for quick response and your kindness. 
Now I understand how batchnorm works in your implementation.

Thanks again for your great work.

Best regards,
Jimyeong
